### ISSUE TITLE
Eventlet threads?

### ISSUE THREAD

**Issue by drozzy** [mrocklin/multipledispatch#51]
Is this safe to use with eventlet threads?

I seem to be getting some weird behavior, and I'm not sure whether it's my code or the multipledispatch.

**Comment by mrocklin** [comment]
I wouldn't trust registration + calling to be threadsafe but, assuming that you've registered all of the signatures that you plan to register I don't see an immediate issue.  I haven't looked at this in depth though.

**Comment by drozzy** [comment]
It seems weird... i'm using pykka (which uses eventlet underneath) and when I send messages from actor to actor, sometimes new actors "override" the old actors.... It's almost as if multipledispatch is "redirecting" messages.

Sorry I know it's vague, just trying to see if there could be a problem with this. I'll try taking it out and comparing the behavior.

**Comment by drozzy** [comment]
Yeah... it seems multipledispatch is doing something. When remove it and replace stuff with "isinstance" checks it works fine.

So I guess be careful with this when using eventlet :-)

**Comment by mrocklin** [comment]
Are you serializing the dispatchers and sending them to different machines or is this all in shared memory?

**Comment by drozzy** [comment]
This is on one machine... same process.
